pkg/cmd/unpack/unpack.go

```diff
@@ -145,10 +161,18 @@ func runUnpackRecursive(ctx context.Context, opts *unpackOptions, visitedRefs []
 			docsIdx += 1
 		}
 
-		if err := unpackLayer(ctx, store, layerDesc, relPath, opts.overwrite, mediaType.Compression); err != nil {
+		// Run the unpack operation in a goroutine
+		wg.Add(1)
+		go unpackLayerAsync(layerDesc, relPath, mediaType)
+	}
+	wg.Wait()
+	close(errChan)
+	for err := range errChan {
+		if err != nil {
```

Comment:
Better to capture all errors, you can combine them using `errors.Join`

Reply:
This change has been implemented